libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationService.java

```diff
@@ -286,13 +286,7 @@ private void startForegroundNotification(Notification notification, int notifica
    */
   @SuppressWarnings("MissingPermission")
   private boolean isValidLocationUpdate(Location location) {
-    if (location == null) {
-      return false;
-    }
-    // If the locations the same as previous, no need to recalculate things
-    return !(location.equals(locationEngine.getLastLocation())
-      || (location.getSpeed() <= 0 && location.hasSpeed())
-      || location.getAccuracy() >= 100);
+    return location != null && location.hasSpeed() && location.getAccuracy() <= 100;
```

Comment:
Magic number `100`. `Location#getAccuracy` returns `the estimated horizontal accuracy of this location, radial, in meters` what about extracting a constant and give it a name based on what `getAccuracy` Javadoc states?